lib/utils/registry/registry_windows.go

```diff
@@ -0,0 +1,100 @@
+//go:build windows
+
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package registry
+
+import (
+	"errors"
+	"os"
+	"strconv"
+
+	"golang.org/x/sys/windows/registry"
+
+	"github.com/gravitational/trace"
+	log "github.com/sirupsen/logrus"
+)
+
+// GetOrCreateRegistryKey loads or creates a registry key handle and passes it back
+func GetOrCreateRegistryKey(name string) (registry.Key, error) {
+	reg, err := registry.OpenKey(registry.CURRENT_USER, name, registry.QUERY_VALUE|registry.CREATE_SUB_KEY|registry.SET_VALUE)
+	if errors.Is(err, os.ErrNotExist) {
+		log.Debugf("Registry key %v doesn't exist, trying to create it", name)
+		reg, _, err = registry.CreateKey(registry.CURRENT_USER, name, registry.QUERY_VALUE|registry.CREATE_SUB_KEY|registry.SET_VALUE)
+		if err != nil {
+			log.Debugf("Can't create registry key %v: %v", name, err)
+			return reg, err
+		}
+	} else {
+		log.Errorf("registry.OpenKey returned error: %v", err)
+		return reg, err
+	}
```

Comment:
This might read better with a switch.

```
switch {
case errors.Is(err, os.ErrNotExist):
    // ...
case err != nil:
    return reg, err
default:
    return reg, nil
}
```

Reply:
Nice!